admin-panel/src/main/java/net/cryptic_game/backend/admin/endpoints/website_management/faq/FAQEntryEndpoints.java

```diff
@@ -0,0 +1,87 @@
+    private final Pattern xxsFilter = Pattern.compile("<[^- ]|[^- ]>");
+
+    @ApiEndpoint(id = "change_answer", authentication = Permission.FAQ_MANAGEMENT)
+    public ApiResponse changeAnswer(@ApiParameter(id = "element_id") final UUID elementId,
+                                    @ApiParameter(id = "new_answer") final String newAnswer) {
+        final FAQEntry entry = this.faqEntryRepository.findById(elementId).orElse(null);
+        if (entry == null) {
+            return new ApiResponse(HttpResponseStatus.BAD_REQUEST, "NO_SUCH_ELEMENT");
+        }
+        if (checkXXS(newAnswer)) {
+            return new ApiResponse(HttpResponseStatus.BAD_REQUEST, "NO_HTML_TAGS_ALLOWED");
+        }
+        entry.setAnswer(newAnswer);
+        return new ApiResponse(HttpResponseStatus.OK, this.faqEntryRepository.saveAndFlush(entry));
+    }
+
+    @ApiEndpoint(id = "create", authentication = Permission.FAQ_MANAGEMENT)
+    public ApiResponse createAnswer(@ApiParameter(id = "question") final String question,
+                                    @ApiParameter(id = "answer") final String answer) {
+        if (question == null || answer == null) {
+            return new ApiResponse(HttpResponseStatus.BAD_REQUEST, "PARAMETERS_CANNOT_BE_NULL");
+        }
+
+        if (this.faqEntryRepository.getByQuestion(question).isPresent()) {
+            return new ApiResponse(HttpResponseStatus.BAD_REQUEST, "QUESTION_ALREADY_EXISTS");
+        }
+
+        if (this.checkXXS(question) || this.checkXXS(answer)) {
+            return new ApiResponse(HttpResponseStatus.BAD_REQUEST, "NO_HTML_TAGS_ALLOWED");
+        }
+
+        final FAQEntry faqEntry = new FAQEntry(question, answer);
+        return new ApiResponse(HttpResponseStatus.OK, this.faqEntryRepository.saveAndFlush(faqEntry));
+    }
+
+    @ApiEndpoint(id = "delete", authentication = Permission.FAQ_MANAGEMENT)
+    public ApiResponse deleteEntry(@ApiParameter(id = "element_id") final UUID elementId) {
+        final Optional<FAQEntry> faqEntryOpt = this.faqEntryRepository.findById(elementId);
+        if (faqEntryOpt.isPresent()) {
+            this.faqEntryRepository.delete(faqEntryOpt.get());
+            return new ApiResponse(HttpResponseStatus.OK);
+        } else {
+            return new ApiResponse(HttpResponseStatus.BAD_REQUEST, "NO_SUCH_ELEMENT");
+        }
+    }
+
+    @ApiEndpoint(id = "list")
+    public ApiResponse list() {
+        final List<FAQEntry> faqEntryList = this.faqEntryRepository.findAll();
+        return new ApiResponse(HttpResponseStatus.BAD_REQUEST, faqEntryList);
+    }
+
+    /**
+     * returns true if an xxs attempt might be there.
+     *
+     * @param content content to prove
+     * @return true if html tags might be in the content
+     */
+    private boolean checkXXS(final String content) {
+        //RegEx to find everything with < and no space or minus after it and > with not space or minus before
+        return this.xxsFilter.matcher(content).find();
```

Comment:
If you use my regex you can uses `matches()`

Reply:
I tried it, it didn't worked